app/frontend/wallet/shelley-wallet.ts

```diff
@@ -295,61 +288,75 @@ const ShelleyWallet = ({config, randomInputSeed, randomChangeSeed, cryptoProvide
   }
 
   async function getMaxNonStakingAmount(address) {
-    const utxos = (await getUTxOs()).filter(isUtxoNonStaking)
+    const utxos = (await getUTxOs()).filter(({address}) => !isGroup(address))
     return _getMaxSendableAmount(utxos, address, false, 0, false)
   }
 
-  const uTxOTxPlanner = async (args, txType) => {
-    const {address, coins, donationAmount, pools} = args
-    const utxoFilters = {
-      delegation: isShelleyUtxo,
-      nonStakingConversion: isUtxoNonStaking,
-      utxo: () => true,
-    }
-    const utxoFilter = utxoFilters[txType]
-    const accountAddress = await myAddresses.accountAddrManager._deriveAddress(0)
-    const availableUtxos = (await getUTxOs()).filter(utxoFilter)
+  type utXoArgs = {
+    address?: string
+    donationAmount?: Lovelace
+    coins?: Lovelace
+    pools?: any
+    txType?: string
+  }
+
+  const uTxOTxPlanner = async (args: utXoArgs, accountAddress: string) => {
+    const {address, coins, donationAmount, pools, txType} = args
     const changeAddress = await getChangeAddress()
-    // we do it pseudorandomly to guarantee fee computation stability
+    const availableUtxos = await getUTxOs()
+    const nonStakingUtxos = availableUtxos.filter(({address}) => !isGroup(address))
+    const groupAddressUtxos = availableUtxos.filter(({address}) => isGroup(address))
     const randomGenerator = PseudoRandom(seeds.randomInputSeed)
-    const shuffledUtxos = shuffleArray(availableUtxos, randomGenerator)
+    const shuffledUtxos =
+      txType === 'convert'
+        ? shuffleArray(nonStakingUtxos, randomGenerator)
+        : [
```

Comment:
is it really intended to shuffle non staking and group address utxos separately, doesn't it hamper the randomness? I would find more straightforward (and randomized) to shuffle the whole array

Reply:
Yes, it is. At least we 've talked about it at the meeting, the plan was to spend non-staking utxos first. One reason is that non-staking utxos are not something you want to possess (in most cases) and the second one is that spending some coins from non-staking balance, and some from staking-balance might be confusing for users. But we can definitely change this, it doesn't seem that crucial to me.

Reply:
I see - that's worth a comment in the code then for future contributors